src/blockdata/transaction.rs

```diff
@@ -673,10 +673,15 @@ impl fmt::Display for NonStandardSigHashType {
 #[cfg_attr(docsrs, doc(cfg(feature = "std")))]
 impl error::Error for NonStandardSigHashType {}
 
+/// Legacy Hashtype of an input's signature
+#[deprecated(since="0.28.0", note="Please use [`LegacySigHashType`] instead")]
+pub type SigHashType = LegacySigHashType;
+
 /// Hashtype of an input's signature, encoded in the last byte of the signature
 /// Fixed values so they can be casted as integer types for encoding
+/// See also [`crate::SchnorrSigHashType`]
 #[derive(PartialEq, Eq, Debug, Copy, Clone)]
-pub enum SigHashType {
+pub enum LegacySigHashType {
```

Comment:
nit: Why not `EcdsaSigHashType`?  One still have to use this type in all non-BIP-340 contexts, so not sure whether it is actually "legacy".

Reply:
Maybe you can rename also the commit message with `EcdsaSigHashType`